File: tools/probe-server.js
// Simple HTTP server to receive iCUE API Probe results
// Usage: node probe-server.js
// Listens on http://localhost:9876
// Results are printed to console and saved to api-probe-results.txt

const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = 9876;
const OUTPUT_FILE = path.join(__dirname, 'api-probe-results.txt');

const server = http.createServer(function(req, res) {
    // CORS headers so the widget can POST from iCUE's webview
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    if (req.method === 'OPTIONS') {
        res.writeHead(204);
        res.end();
        return;
    }

    if (req.method === 'POST' && req.url === '/probe-results') {
        var body = '';
        req.on('data', function(chunk) { body += chunk; });
        req.on('end', function() {
            console.log('\n' + '='.repeat(60));
            console.log('Probe results received at ' + new Date().toISOString());
            console.log('='.repeat(60));
            console.log(body);
            console.log('='.repeat(60) + '\n');

            fs.writeFileSync(OUTPUT_FILE, body, 'utf8');
            console.log('Saved to ' + OUTPUT_FILE);

            res.writeHead(200, { 'Content-Type': 'text/plain' });
            res.end('OK');
        });
        return;
    }

    res.writeHead(404);
    res.end('Not found');
});

server.listen(PORT, function() {
    console.log('Probe server listening on http://localhost:' + PORT);
    console.log('Waiting for API Probe widget to POST results...');
    console.log('Results will be saved to ' + OUTPUT_FILE);
});

File: tools/probe_server.py
"""
Simple HTTP server that receives API Probe results from the iCUE widget.
Run this, then load the ApiProbe widget in iCUE — it will auto-POST results.

Usage:
    python probe_server.py

Results are saved to probe-results.txt in the current directory.
"""

from http.server import BaseHTTPRequestHandler, HTTPServer


class Handler(BaseHTTPRequestHandler):
    def do_POST(self):
        length = int(self.headers.get("Content-Length", 0))
        body = self.rfile.read(length)

        with open("probe-results.txt", "wb") as f:
            f.write(body)

        self.send_response(200)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.end_headers()
        self.wfile.write(b"OK")
        print(f"\nSaved {length} bytes to probe-results.txt")

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Access-Control-Allow-Methods", "POST, OPTIONS")
        self.send_header("Access-Control-Allow-Headers", "Content-Type")
        self.end_headers()


if __name__ == "__main__":
    server = HTTPServer(("", 9876), Handler)
    print("Listening on http://localhost:9876")
    print("Load the ApiProbe widget in iCUE — results will auto-POST here.")
    print("Press Ctrl+C to stop.\n")
    try:
        server.serve_forever()
    except KeyboardInterrupt:
        print("\nStopped.")
        server.server_close()
